### ISSUE TITLE
Hasura metadata apply fails when trying to apply exported metadata with remote_schema permissions

### ISSUE THREAD

**Issue by AesSedai** [hasura/graphql-engine#7608]
### Version Information

Server Version: hasura/graphql-engine:v2.0.8.cli-migrations-v3
CLI Version (for CLI related issue): v2.0.0-beta.2

### Environment
OSS

### What is the expected behaviour?
Hasura is unable to apply the metadata for a remote_schema it itself has exported. Hasura should be able to apply its own exported metadata.

### Keywords
metadata, remote schema, remote schema permission, remote_schema, remote_schema_permission

### What is the current behaviour?
Hasura exports the metadata successfully, but then fails to apply it because of a parsing / endOfInput error in the `user` role in the `metadata/remote_schemas.yaml` file:
```
# ~/et2/hasura on git:main x [22:35:04] C:1
$ hasura metadata apply --admin-secret <XXXXXX>
INFO Metadata applied                             

# ~/et2/hasura on git:main x [22:35:16] 
$ hasura metadata apply --admin-secret <XXXXXX>
FATA[0000] error applying metadata 
{
  "path": "$.args.metadata.remote_schemas[0].permissions[0].definition.schema",
  "error": "parsing the schema document: \"endOfInput\"",
  "code": "parse-failed"
} 
```

### How to reproduce the issue?

1. Try to apply the attached metadata?

### Please provide any traces or logs that could help here.
Hasura metadata that exhibits the issue. Please refer to the `remote_schemas.yaml` file.
[hasura.zip](https://github.com/hasura/graphql-engine/files/7249128/hasura.zip)

### Any possible solutions?
I've been deleting the `remote_schemas.yaml` file and using the metadata API to import the remote schema and permissions, but that shouldn't be required.

EDIT: with assistance from benoit, it was determined that the issue is related to extra random backslashes present in the exported `remote_schemas.yaml` file. This newly provided file with the backslashes removed can be applied correctly:
[remote_schemas.zip](https://github.com/hasura/graphql-engine/files/7249321/remote_schemas.zip)

### Can you identify the location in the source code where the problem exists?
The metadata apply works when the `user` role is removed from the `remote_schemas.yaml` file, so it's something to do with how the `user` role schema is parsed.

### If the bug is confirmed, would you be willing to submit a PR?
No, I don't know haskell :(

**Comment by scriptonist** [comment]
@AesSedai can you verify if you can reproduce this on the latest version of CLI?
`hasura update-cli --version v2.0.9` should help you update to the latest version.

**Comment by AesSedai** [comment]
@scriptonist I updated the CLI and still encountered the same issue when trying to export / apply. I edited the above issue, but Benoit helped me narrow down the issue to improperly added backslashes present in the `user` role in the `remote_schemas.yaml` file. Attached in the original issue is a zip with the updated `remote_schemas.yaml` that can be applied correctly.

This narrows the issue down to how hasura exports the remote schema, wherein it inserts incorrect linebreaks?

**Comment by RobSchilderr** [comment]
@AesSedai have you fixed this in the end?

**Comment by AesSedai** [comment]
@RobSchilderr I've worked around it, haven't checked to see if the CLI has been updated. Basically, I just used regex to remove the extra backslashes and that allowed it to be imported.

**Comment by RobSchilderr** [comment]
> @RobSchilderr I've worked around it, haven't checked to see if the CLI has been updated. Basically, I just used regex to remove the extra backslashes and that allowed it to be imported. 

Aah I am having a similar error in the exact same version when merging remotely through the console.

`Saving permission failed: `{
    "path": "$.args[0].args.definition.schema",
    "error": "parsing the schema document: \"endOfInput\"",
    "code": "parse-failed"
}`

**Comment by rscheuermann** [comment]
+1 We are also encountering this issue. If we let Hasura write the remote_schemas.yaml with permissions, it's unable to be parsed on startup due to the extra backslashes

```{"path":"$.args.metadata.remote_schemas[0].permissions[0].definition.schema","error":"parsing the schema document: \"endOfInput\"","code":"parse-failed"}```

**Comment by vitaliytv** [comment]
my code for temporary fix:

```javascript
import yaml from 'js-yaml'
import { readFileSync, writeFileSync } from 'fs'

const file = './metadata/remote_schemas.yaml'

const doc = yaml.load(readFileSync(file, 'utf8'))

writeFileSync(file, yaml.dump(doc, { noArrayIndent: true }))
```

**Comment by john-mcinall** [comment]
This issue is not just related to the extra backslashes, I can replicate this issue with a schema file that doesn't have the extra back slashses at the start of the lines.

I have noticed at times it can be cause but the location of an enum delaration or simply a space missing before a bracket to name a few things that I changed to make it work. But all in all this is a horrible experience when using the CLI to export and import metadata top hasura. 

I have lost numerous hours just trying to figure out what the actual syntax error is in the generated schema.

Is there any update about the root cause of this issue and if it is even activley being looked into?

**Comment by raphaeltm** [comment]
I just want to add that I have wasted quite a few hours over the past few days trying to figure out why deployments were failing. I figured out it had to do with enums in the remote schema and eventually found this thread.

What's funny is I can put the yaml through a linter that validates it is okay (and changes the schema from the double quote syntax to the `|-` syntax), and then pass the schema through a GraphQL linter, and then put it back into the `remote_schemas.yaml` file, and everything works as it should.
